From c09486432072ab9f0b1fff62089fea67a0cc2573 Mon Sep 17 00:00:00 2001
From: Aleksey Makarov <amakarov@marvell.com>
Date: Mon, 8 Oct 2018 23:58:51 +0300
Subject: [PATCH 0233/1051] net: octeontx2: Stop NAPI polling before resetting

Stop NAPI polling before resetting.  Without this
queue interrupts will happen again and again.

Signed-off-by: Aleksey Makarov <amakarov@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
index 1414f3f9c7ff..27766b65b524 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
@@ -278,11 +278,13 @@ int otx2_napi_handler(struct otx2_cq_queue *cq,
 	cq_status = otx2_nix_cq_op_status(pfvf, cq->cq_idx);
 	if (cq_status & BIT_ULL(63)) {
 		dev_err(pfvf->dev, "CQ operation error");
+		pfvf->intf_down = true;
 		schedule_work(&pfvf->reset_task);
 		return 0;
 	}
 	if (cq_status & BIT_ULL(46)) {
 		dev_err(pfvf->dev, "CQ stopped due to error");
+		pfvf->intf_down = true;
 		schedule_work(&pfvf->reset_task);
 		return 0;
 	}
-- 
2.17.1

